### PR TITLE
port text_viewer from gtk-rs/examples4

### PR DESCRIPTION
**Pull request by vamsikrishna-brahmajosyula**

### CODE REVIEW

File: examples/src/bin/text_viewer.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//! # GtkBox, Scrollable Text View and File Chooser
+//!
+//! A simple text file viewer
+
+use gio::prelude::*;
```
Comment (bilelmoussaoui) [inline examples/src/bin/text_viewer.rs:5]:
```suggestion
```
shouldn't be needed anymore

File: examples/src/bin/ui/text_viewer.ui [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <property name="default_width">400</property>
+    <property name="default_height">480</property>
+    <child>
+      <object class="GtkBox" id="v_box">
+        <property name="orientation">vertical</property>
+        <child>
+          <object class="GtkBox">
+            <child>
+              <object class="GtkButton" id="open_button">
+                <property name="visible">1</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/text_viewer.ui:14]:
```suggestion
```

Comment (bilelmoussaoui) [reply examples/src/bin/ui/text_viewer.ui:14]: widgets are visible by default

File: examples/src/bin/ui/text_viewer.ui [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                <property name="label" translatable="yes">Open</property>
+                <property name="icon-name">document-open-symbolic</property>
+                <property name="tooltip-text" translatable="yes">Open</property>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="GtkScrolledWindow" id="scrolled_window">
+            <property name="can_focus">1</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/text_viewer.ui:24]:
```suggestion
```

Comment (bilelmoussaoui) [reply examples/src/bin/ui/text_viewer.ui:24]: the default value is properly set already

File: examples/src/bin/ui/text_viewer.ui [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            <property name="hexpand">1</property>
+            <property name="vexpand">1</property>
+            <child>
+              <object class="GtkTextView" id="text_view">
+                <property name="can_focus">1</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/text_viewer.ui:29]:
```suggestion
```

File: examples/src/bin/ui/text_viewer.ui [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                <property name="tooltip-text" translatable="yes">Open</property>
+              </object>
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="GtkScrolledWindow" id="scrolled_window">
+            <property name="can_focus">1</property>
+            <property name="hexpand">1</property>
+            <property name="vexpand">1</property>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/text_viewer.ui:26]:
```suggestion
            <property name="hexpand">True</property>
            <property name="vexpand">True</property>
```
Those are booleans.

File: examples/src/bin/text_viewer.rs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+        // TODO move this to a impl?
+        let file_chooser = FileChooserDialog::new(
+            Some("Open File"),
+            Some(&window),
+            FileChooserAction::Open,
+            &[],
+        );
+
+        file_chooser.add_buttons(&[("Open", ResponseType::Ok), ("Cancel", ResponseType::Cancel)]);
```
Comment (bilelmoussaoui) [inline examples/src/bin/text_viewer.rs:44]: Why add the buttons here instead of directly when creating the file chooser?

**Comment by vamsikrishna-brahmajosyula** [comment]
I still need to learn how to rebase these feature branches on upstream, sending a new pull request with comments addressed.

**Comment by sdroege** [comment]
Maybe https://git-scm.com/book/en/v2/Git-Branching-Rebasing helps?

**Comment by vamsikrishna-brahmajosyula** [comment]
thanks, will go through.

**Pull request closed without merging**